src/simulations/collect.jl

```diff
@@ -0,0 +1,246 @@
+"""
+Collects agent properties (fields of the agent object) into a dataframe.
+"""
+function collect_agent_data(model::ABM, properties::Array{Symbol}; step=0)
+  dd = DataFrame()
+  dd[!, :id] = collect(keys(model.agents))
+  for fn in properties
+    dd[!, fn] = getproperty.(values(model.agents), fn)
+  end
+  dd[!, :step] = repeat([step], size(dd, 1))
+  return dd
+end
+
+"""
+Collects agent properties (fields of the agent object) into a dataframe
+and appends them to the supplied `df`.
+"""
+function collect_agent_data!(df::DataFrame, model::ABM, properties::Array{Symbol}; step::Integer)
```

Comment:
If `step` is mandatory argument, why is it a keyword? It probably makes sense to change `step` from a keyword to argument. The same for the above method, `; step = 0` -> `, step = 0`.